gsi/gridmap_eppn_callout/globus_gridmap_eppn.c

```diff
@@ -615,24 +616,45 @@ globus_gridmap_eppn_callout(
         context, subject, &found_identity, shared_user_cert, shared_user_chain, shared_exp);
     if(result == GLOBUS_SUCCESS)
     {
-        if(desired_identity && strcmp(found_identity, desired_identity) != 0)
+        if(desired_identity == NULL)
         {
-            GLOBUS_GRIDMAP_CALLOUT_ERROR(
-                result,
-                GLOBUS_GRIDMAP_CALLOUT_LOOKUP_FAILED,
-                ("Credentials specify id of %s, can not allow id of %s.\n",
-                 found_identity, desired_identity));
-            globus_free(found_identity);
-            goto error;
+            if(getenv("GLOBUS_EPPN_FORCE_GRIDMAP"))
+            {
+                result = globus_gss_assist_gridmap(found_identity, &tmp_identity);
+                if(result != 0)
```

Comment:
globus_gss_assist_gridmap returns int. "result" is a globus_result_t, so this does not match. Should be
rc = ...
if (rc != 0)

Reply:
Ah yes, you're totally right. So it should be a different variable indeed. But then the question is what to pass to `GLOBUS_GRIDMAP_CALLOUT_ERROR()`:
Going through the whole file, I notice that `GLOBUS_GRIDMAP_CALLOUT_ERROR()` is often called with `result == GLOBUS_SUCCESS`, e.g. in `ggvm_load_cert_from_file()`, while in other cases it's actually initialized to `GLOBUS_FAILURE`, as in `ggvm_verify_cert()` ?! I'd say `GLOBUS_FAILURE` is the logical one, but not in line with what most of the rest of the file does.
The second case below is easier, there we still have the top error in `result`

Reply:
Calling GLOBUS_GRIDMAP_CALLOUT_ERROR with GLOBUS_SUCCESS as input is correct if the error you are reporting is the original error and isn't caused by another error.
Calling GLOBUS_GRIDMAP_CALLOUT_ERROR with GLOBUS_FAILURE as input will create a confusing error message stating that the error you are reporting was caused by a different unknown error - which is not true.